src/App.js

```diff
@@ -1,25 +1,55 @@
-import logo from './logo.svg';
+import React, { Component } from 'react';
 import './App.css';
+import Movies from './Movies.js';
+import MovieDetails from './MovieDetails';
+import Nav from './Nav.js';
+import { getMovies } from './apiCalls';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends Component {
+  constructor() {
+    super();
+    this.state = {
+      movies: [],
+      isSingleMovie: false,
+      error: '',
+    };
+  }
+
+  componentDidMount() {
+    getMovies()
+      .then((data) =>
+        this.setState({ movies: [...this.state.movies, ...data.movies] })
+      )
+      .catch((error) =>
+        this.setState({ error: 'We are sorry, something went wrong :(' })
+      );
+  }
+
+  handleChange() {
+    return this.setState({ isSingleMovie: true });
+  }
+
+  render() {
+    if (this.state.error) {
+      return <p>{this.state.error}</p>;
+    }
+
+    if (!this.state.error && !this.state.isSingleMovie) {
+      return (
+        <div className="app">
+          <Nav />
+          <Movies
+            movies={this.state.movies}
+            handleChange={this.state.handleChange}
+          />
+        </div>
+      );
+    }
+
+    if (!this.state.error && this.state.isSingleMovie) {
+      return <MovieDetails />;
+    }
+  }
```

Comment:
We typically want to avoid having logic inside of our render, but I'm guessing this has been refactored to include router at this point anyway - Let me know if you have any questions about this!